src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -39,32 +39,22 @@ public abstract class ResolveOperationBase : IResolveOperation, ITracingIdentife
         private const int SuccessListInitialCapacity = 32;
 
         private bool _ended;
-        private IResolvePipelineTracer? _pipelineTracer;
-        private List<ResolveRequestContext> _successfulRequests = new List<ResolveRequestContext>(SuccessListInitialCapacity);
+        private readonly IResolvePipelineTracer? _pipelineTracer;
+        private readonly List<ResolveRequestContext> _successfulRequests = new List<ResolveRequestContext>(SuccessListInitialCapacity);
         private int _nextCompleteSuccessfulRequestStartPos = 0;
 
-        /// <summary>
```

Comment:
I have mixed feelings about the removal of the constructor that has no tracer. I get that you can just do the optional parameter, but I also like the semantics of there being two different 'modes' of tracing.

I think this may change with @tillig's DiagnosticSource changes anyway, so we can revisit that when those changes are ready.

Reply:
With DiagnosticSource, it's sort of a global singleton at the container level. It gets passed down from lifetime scope to lifetime scope, but it's not as flexible as having a tracer that only sits at a single scope level. On the other hand, it's super efficient and you can subscribe to individual events pretty easy for things like perf counters if you want.

Net: the constructor parameter with the tracer goes away and the DiagnosticSource on the scope gets used. I wouldn't worry about messing with this too much.

Reply:
What's the advantage of the constructor that I've removed, when it also went and called `this(mostNestedLifetimeScope, null)` anyway?

Reply:
It's mostly just a mild style preference; it's sometimes nicer to just have separate constructors, especially if you might want different summary comments (which I didn't actually avail myself of here).

I don't feel strongly about it either way, and it sounds like it'll be changed anyway with the DiagnosticSource updates.